cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -33,6 +33,13 @@ type MachineTypeKey struct {
 	MachineTypeName string
 }
 
+// InstanceTemplateNameType is used to store the name, and
+// whether or not the instance template is regional
+type InstanceTemplateNameType struct {
+	Name     string
+	Regional bool
```

Comment:
I think you will need to store the region here as well, otherwise it is not possible to discern if 2 template names are the same. Please also add a dedicated `Equal` function that compares 2 template names.

Reply:
Will add the Equal function. No need to store the region, as the instance template name is unique. So two instance templates of the same nodepool but of different regions will have different names.